src/network/networkd-address.c

```diff
@@ -2179,6 +2212,19 @@ static int address_section_verify(Address *address) {
             !FLAGS_SET(address->duplicate_address_detection, ADDRESS_FAMILY_IPV6))
                 address->flags |= IFA_F_NODAD;
 
+        uint32_t filtered_flags = address->family == AF_INET ?
+                address->flags & KNOWN_FLAGS & ~UNMANAGED_FLAGS & ~IPV6ONLY_FLAGS :
+                address->flags & KNOWN_FLAGS & ~UNMANAGED_FLAGS;
+        if (address->flags != filtered_flags) {
```

Comment:
doesn't this mean that every time the kernel is updated, networkd needs to be updated too? Isn't that a bit too unflexible?

Reply:
This verifies [Address] settings in .network files, not verifying the addresses assigned to the system.
So, unless we add more knobs to configure flags, the logic is not necessary to be updated, even if the future kernel adds more flags.